### ISSUE TITLE
[Question] Gradle, Scope & Kotlin

### ISSUE THREAD

**Issue by fb33** [oss-review-toolkit/ort#4299]
Hi,

I did an audit for a project using Gradle and Kotlin.
The scan step (with scanCode) takes a huge time for some dependencies not in cache.
Some of theses dependencies looks like internal compiler deps, example : Maven:org.jetbrains.kotlin:kotlin-daemon-embeddable:1.5.10 or Maven:org.jetbrains.kotlin:kotlin-scripting-compiler-embeddable:1.5.10 and are defined in specific scope like kotlinCompilerClasspath.

I don't have this kind of dependency with other projects using Maven and Kotlin.

My question is: can we consider that scopes in Kotlin* (kotlinCompilerClasspath, kotlinCompilerPluginClasspath et kotlinKlibCommonizerClasspath) to be excluded? If so, maybe they should be added in the configuration examples ?

Regards

**Comment by tsteenbe** [comment]
@fb33 Yes, we commonly exclude some of the scopes you mentioned for Gradle project see for example https://github.com/oss-review-toolkit/ort/issues/4294, this file is part of ORT's helper CLI.

You can generate the scope excludes in an .ort.yml using the helper CLI by running the following command:

```
[ort_home]/helper-cli/build/install/orth/bin/orth repository-configuration generate-scope-excludes \
    --ort-file [analyzer-result.json or scan-result.json file] \
    --repository-configuration-file [.ort.yml file]
```

**Comment by sschuberth** [comment]
Actually, my recommended to speed up the process for such "basic" framework / SDK / compiler type of dependencies is to create a curation that sets both the concluded license and author for these, and enable the scanner's [`skipConcluded` configuration setting](https://github.com/oss-review-toolkit/ort/blob/309868929e5feee9a353ba520764ca5de9de5ebe/model/src/main/resources/reference.conf#L53). That way, such dependencies won't be scanned or downloaded at all, but will still be attributed for.
